mockOSApi.v2/Models/Task/MockProcessBuilder.cs

```diff
@@ -0,0 +1,133 @@
+using mockOSApi.Repository;
+using mockOSApi.Errors;
+
+namespace mockOSApi.Models;
+
+public interface IMockProcessBuilder
+{
+    public MockProcessBuilder AddPriority(int? priority);
+    public MockProcessBuilder AddPid();
+    public MockProcessBuilder AddProcessStatus();
+    public MockProcessBuilder AddExitCode();
+    public MockProcessBuilder AddDefaultFds();
+    public MockProcessBuilder AddImage(string? image);
+    public MockProcessBuilder AddCreationTime();
+    public MockProcessBuilder AddArguments(string[]? arguments);
+    public MockProcessBuilder AddUserContext(User user);
+    public MockProcess Build();
+
+}
+
+public class MockProcessBuilder : IMockProcessBuilder
```

Comment:
I really like this! Great job on coming up with this.
One suggestion, if you now have this class, you can move the Pid generation logic here, instead of having it on the MockProcess

Reply:
The PID is assigned in this class, see: public MockProcessBuilder AddPid()